src/main/java/de/zalando/aruha/nakadi/partitioning/PartitionsCache.java

```diff
@@ -0,0 +1,38 @@
+package de.zalando.aruha.nakadi.partitioning;
+
+import de.zalando.aruha.nakadi.exceptions.ServiceUnavailableException;
+import de.zalando.aruha.nakadi.repository.kafka.KafkaFactory;
+import kafka.common.KafkaException;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class PartitionsCache {
+
+    private final Map<String, String[]> partitionsPerEventType = new ConcurrentHashMap<>();
+
+    private final KafkaFactory kafkaFactory;
+
+    public PartitionsCache(final KafkaFactory kafkaFactory) {
+        this.kafkaFactory = kafkaFactory;
+    }
+
+    public String[] getPartitionsFor(final String eventTypeName) throws ServiceUnavailableException {
+        try {
+            String[] partitions = partitionsPerEventType.get(eventTypeName);
+
+            if (partitions == null) {
+                partitions = kafkaFactory.createProducer().partitionsFor(eventTypeName)
```

Comment:
This class is too kafka-specific.
I think it would be more correct to abstract from kafka using TopicRepository layer